### ISSUE TITLE
Navigation through history.pushState() is conflicting with <ui-nav-menu> sometimes

### ISSUE THREAD

**Issue by eugals** [Shopify/shopify-app-bridge#394]
I'm trying to migrate my SPA from AppBridge v3 to v4. After applying all the recommendations in #240, I have mostly managed to get my code working in this new environment. However, there is a small, problematic behavior that appears to be a bug in v4.

Here's a short video demonstrating the issue:

https://github.com/Shopify/shopify-app-bridge/assets/479407/fbf9fc54-262a-4ef1-a04e-bb4ffa0b98c6

Here's what's happening:

1. When navigating through clicks on `<ui-nav-menu>`, everything works fine. My `BrowserRouter` receives all the necessary notifications through `Link.onClick`, and the change is reflected within my UI.
2. When navigating through the internal tabs, it initially works fine. My code updates the current path through `history.pushState()` calls, and the menu somehow picks up these changes and updates the current item's selection accordingly.
3. The problem arises when I use both navigation methods. Once you navigate using method (1), it causes method (2) (the pushState-based navigation) to malfunction. The current selection stops being picked up until you refresh the page. This issue is visible around the 20-second mark in the attached video.

Additionally, if I try to `pushState` programmatically, it sometimes causes the entire webpage to reload when I later attempt to navigate via the sidebar.

**Am I doing something wrong, or is this a bug in AppBridge?**

This is how I generate the menu:
```JSX
<ui-nav-menu>
   {s.tabs.map(t => {
        <Link key={t.id} to={t.href} rel={t.accessibilityLabel} onClick={this.onNav.bind(this, t)}>{t.title}</Link>
   })}
</ui-nav-menu>
```
And this is how I navigate:
```JS
history.pushState(null, '', t.href + initialLocationSearch);
```

For comparison here's a short video with the same action recorded on a version of my code still running via AppBridge v3:

https://github.com/Shopify/shopify-app-bridge/assets/479407/08d71721-ba31-4d31-918b-a869a0a0bde3

Everything works perfectly fine through the old, clean and straightforward [NavigationMenu](https://shopify.dev/docs/api/app-bridge/previous-versions/actions/menu/navigation) API. I wish it was still around in v4. It would make everyone's life sooo much easier I believe.

**Comment by eugals** [comment]
> I tried [this](https://shopify.dev/docs/api/app-bridge/previous-versions/actions/navigation/redirect-navigate#redirect-to-a-relative-path-in-the-app) instead of `history.pushState`. It works fine in my case.

Yes, as I already mentioned at the start, that v3 API works flawlessly. Too bad it is not available under AppBridge v4 anymore.

**Comment by vladucu** [comment]
We're also seeing this issue while trying to migrate from AppBridge `v3` -> `v4`. Any updates here?
